modules/standard/FileSystem.chpl

```diff
@@ -16,17 +16,78 @@
+   :proc:`copy` functionality belongs here, as does :proc:`chmod`,
+   :proc:`mkdir`, and :proc:`remove`, for instance.  Also included are
+   operations relating to the current process's state, such as :proc:`umask` or
+   :proc:`~locale.chdir` (an operation which is performed on a specified
+   locale).
+
+ */
+module FileSystem {
+
 use Error;
 
+/* S_IRUSR and the following constants are values of the form
+   S_I[R | W | X][USR | GRP | OTH], S_IRWX[U | G | O], S_ISUID, S_ISGID, or
+   S_ISVTX, where R corresponds to readable, W corresponds to writable, X
+   corresponds to executable, USR and U correspond to user, GRP and G
+   correspond to group, OTH and O correspond to other, directly tied to the C
+   idea of these constants.  They are intended for use when dealing with the
+   permissions of files or directories, such as with :proc:`chmod`,
+   :proc:`getMode`, :proc:`mkdir`, or :proc:`umask`
 
-/* Change the current working directory of the current locale to the specified
-   name. Returns any errors that occurred via an out parameter.
-   err: a syserr used to indicate if an error occurred
-   name: a string indicating a new directory
+   S_IRUSR refers to the user's read permission
+*/
+extern const S_IRUSR: int;
+/* Refers to the user's write permission, see :data:`S_IRUSR` */
+extern const S_IWUSR: int;
+/* Refers to the user's executable permission, see :data:`S_IRUSR` */
+extern const S_IXUSR: int;
+/* Refers to all the user's permissions, see :data:`S_IRUSR` */
+extern const S_IRWXU: int;
+
+/* Refers to the group's read permission, see :data:`S_IRUSR` */
+extern const S_IRGRP: int;
+/* Refers to the group's write permission, see :data:`S_IRUSR` */
+extern const S_IWGRP: int;
+/* Refers to the group's executable permission, see :data:`S_IRUSR` */
+extern const S_IXGRP: int;
+/* Refers to all the permissions of those in this group, see :data:`S_IRUSR` */
+extern const S_IRWXG: int;
 
-   Note: this is not safe within a parallel context.  A chdir call in one task
-   will affect the current working directory of all tasks for that locale.
+/* Refers to everyone else's read permission, see :data:`S_IRUSR` */
+extern const S_IROTH: int;
+/* Refers to everyone else's write permission, see :data:`S_IRUSR` */
+extern const S_IWOTH: int;
+/* Refers to everyone else's executable permission, see :data:`S_IRUSR` */
+extern const S_IXOTH: int;
+/* Refers to all the permissions of everyone else, see :data:`S_IRUSR` */
+extern const S_IRWXO: int;
+
+/* Use of this would cause the user id to be set on execution, see
+   :data:`S_IRUSR`
+*/
+extern const S_ISUID: int;
+/* Use of this would cause the group id to be set on execution, see
+   :data:`S_IRUSR`
+*/
+extern const S_ISGID: int;
+/* This is the sticky bit. See `GNU Permissions
+   <http://www.gnu.org/software/libc/manual/html_node/Permission-Bits.html>`_
+   for what that generally means, and :data:`S_IRUSR` for intended uses, etc.
```

Comment:
I think this reference to GNU Permissions is great and that it would be nice to also include general references about these permissions higher up. And also especially with the ISUID/ISGID bits.